modules/custom/external_comment/external_comment.module

```diff
@@ -25,3 +25,78 @@ function external_comment_query_comment_filter_alter(AlterableInterface $query)
     }
   }
 }
+
+
+
+function external_comment_cron() {
+  // fetch comments
+  $comment_ids = \Drupal::entityQuery('comment')
+    ->condition('entity_type', 'node')
+    ->condition('status', 1)
+    ->execute();
+
+  $comments_data = [];
+  // if comments exist
+  if ($comment_ids) {
+    $comments = \Drupal::entityTypeManager()
+      ->getStorage('comment')
+      ->loadMultiple($comment_ids);
+
+    foreach($comments as $comment) {
+      // Loop over and get fields for published comments
+      if ($comment->getStatus() == 1) {
+        $comments_data[] = [
+          'comment_id' => $comment->id(),
+          'uuid' => $comment->uuid(),
+          'subject' => $comment->getSubject(),
+          'comment_body' => $comment->get('comment_body')->getValue()[0]['value'],
+          'comment_posted_by' => $comment->getAuthorName(),
+          'date_posted' => \Drupal::service('date.formatter')->format($comment->getCreatedTime(), 'html_date')
+        ];
+      }
+    }
+  }
+
+  $header = [
+    'Comment id',
+    'Page uuid',
+    'Subject',
+    'Body',
+    'Posted by',
+    'Date posted'
+  ];
+
+  // export as csv
+  write_to_csv('comments.csv', $comments_data, $header);
+}
+
+
+/**
+ * Generate output file for given data and headers
+ */
+function write_to_csv($filename, $data_to_write, $csv_header) {
```

Comment:
if you have only one function writing to a csv, you may include it in the function itself?